persistence-common/src/main/java/io/stargate/db/datastore/common/AbstractCassandraSchemaConverter.java

```diff
@@ -112,8 +113,8 @@ private Keyspace convertKeyspace(K keyspace) {
     Stream<UserDefinedType> userDefinedTypes = convertUserTypes(name, userTypes(keyspace));
     return Keyspace.create(
         name,
-        tables::iterator,
-        userDefinedTypes::iterator,
+        tables.collect(Collectors.toList()),
```

Comment:
To make it even less error-prone, I suggest to not use `Stream` as a variable at all, so this code:
```
    String name = keyspaceName(keyspace);
    Stream<Table> tables = convertTables(name, tables(keyspace), views(keyspace));
    Stream<UserDefinedType> userDefinedTypes = convertUserTypes(name, userTypes(keyspace));
    return Keyspace.create(
        name,
        tables.collect(Collectors.toList()),
        userDefinedTypes.collect(Collectors.toList()),
        replicationOptions(keyspace),
        Optional.of(usesDurableWrites(keyspace)));
  }
```
would be:
```
    String name = keyspaceName(keyspace);
    List<Table> tables = convertTables(name, tables(keyspace), views(keyspace)).collect(Collectors.toList());
    List<UserDefinedType> userDefinedTypes = convertUserTypes(name, userTypes(keyspace)).collect(Collectors.toList());
    return Keyspace.create(
        name,
        tables,
        userDefinedTypes,
        replicationOptions(keyspace),
        Optional.of(usesDurableWrites(keyspace)));
  }
```
having the latter solution, we won't risk that someone else will pass this Stream as an argument in the future.
(same for other occurrences of this pattern)

Reply:
Well, it's not necessary wrong to use streams as arguments, depends what's you're trying to do. The care is only when you need to convert a stream to an iterable, and I could make the counter-argument that the current code does the conversion exactly when it's needed. I think an argument can be made that it's more self-documented as it is than if when-we-convert is detached from why-we-need-to-convert.

Reply:
sure, up to you :) But in this case, we are not reusing `Stream` anywhere (not chaining any subsequent operations) so maybe it is not needed to keep the type.